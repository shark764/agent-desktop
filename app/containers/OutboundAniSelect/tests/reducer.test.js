/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */
import { fromJS } from 'immutable';
import {
  SET_OUTBOUND_IDENTIFICATION_LISTS,
  SELECT_OUTBOUND_IDENTIFICATION,
} from '../constants';
import OutboundAniSelectReducer from '../reducer';

describe('OutboundAniSelectReducer', () => {
  it('returns the initial state', () => {
    expect(OutboundAniSelectReducer(undefined, {})).toMatchSnapshot();
  });

  describe('SET_OUTBOUND_IDENTIFICATION_LISTS', () => {
    it('set outbound identifier list', () => {
      const action = {
        type: SET_OUTBOUND_IDENTIFICATION_LISTS,
        response: {
          result: {
            effective: 'mock-value',
          },
        },
      };
      expect(OutboundAniSelectReducer(new Map(), action)).toMatchSnapshot();
    });
    it('if outbound identifier list are empty', () => {
      const initialState = fromJS({});
      const action = {
        type: SET_OUTBOUND_IDENTIFICATION_LISTS,
        response: {
          result: [],
        },
      };
      expect(
        OutboundAniSelectReducer(fromJS(initialState), action)
      ).toMatchSnapshot();
    });
  });
  describe('SELECT_OUTBOUND_IDENTIFICATION', () => {
    it('set outbound identifier when outbound is selected', () => {
      const action = {
        type: SELECT_OUTBOUND_IDENTIFICATION,
        response: 'mock-value',
      };
      expect(OutboundAniSelectReducer(new Map(), action)).toMatchSnapshot();
    });
    it('if outbound identifier does not selected', () => {
      const initialState = fromJS({});
      const action = {
        type: SELECT_OUTBOUND_IDENTIFICATION,
        response: [],
      };
      expect(
        OutboundAniSelectReducer(fromJS(initialState), action)
      ).toMatchSnapshot();
    });
  });
});
