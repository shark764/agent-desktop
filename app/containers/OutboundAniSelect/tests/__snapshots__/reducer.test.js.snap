// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutboundAniSelectReducer SELECT_OUTBOUND_IDENTIFICATION set an outbound identifier when an outbound identifier email is selected 1`] = `
Immutable.Map {
  "selectedEmailOutboundIdentifier": "mock-value",
}
`;

exports[`OutboundAniSelectReducer SELECT_OUTBOUND_IDENTIFICATION set null when there's no outbound identifier selected 1`] = `
Immutable.Map {
  "selectedEmailOutboundIdentifier": null,
}
`;

exports[`OutboundAniSelectReducer SELECT_OUTBOUND_PHONE_IDENTIFICATION set an outbound identifier when an outbound identifier phone is selected 1`] = `
Immutable.Map {
  "selectedPhoneOutboundIdentifier": "mock-value",
}
`;

exports[`OutboundAniSelectReducer SELECT_OUTBOUND_PHONE_IDENTIFICATION set null when there's no outbound identifier selected 1`] = `
Immutable.Map {
  "selectedPhoneOutboundIdentifier": null,
}
`;

exports[`OutboundAniSelectReducer SET_OUTBOUND_IDENTIFICATION_LISTS if outbound identifier list are empty 1`] = `
Immutable.Map {
  "outboundIdentifierLists": Immutable.List [],
}
`;

exports[`OutboundAniSelectReducer SET_OUTBOUND_IDENTIFICATION_LISTS set outbound identifier list 1`] = `
Immutable.Map {
  "outboundIdentifierLists": Immutable.Map {
    "effective": "mock-value",
  },
}
`;

exports[`OutboundAniSelectReducer returns the initial state 1`] = `
Immutable.Map {
  "outboundIdentifierList": undefined,
  "selectedEmailOutboundIdentifier": undefined,
  "selectedPhoneOutboundIdentifier": undefined,
}
`;
